src/cml.js

```diff
@@ -159,9 +182,63 @@ class CML {
       ? ''
       : '![CML watermark](https://raw.githubusercontent.com/iterative/cml/master/assets/watermark.svg)';
 
-    const report = `${userReport}\n\n${watermark}`;
+    let userReport = testReport;
+    try {
+      if (!userReport)
+        userReport = await fs.promises.readFile(markdownFile, 'utf-8');
+    } catch (err) {
+      if (!watch) throw err;
+    }
+
+    let report = `${userReport}\n\n${watermark}`;
     const drv = getDriver(this);
 
+    const publishLocalFiles = () => async (tree) => {
+      const nodes = [];
+
+      visit(tree, ['definition', 'image', 'link'], (node) => nodes.push(node));
+
+      const visitor = async (node) => {
+        if (node.url && node.alt !== 'CML watermark') {
+          if (!triggerFile) watcher.add(node.url);
+          try {
+            const url = new URL(
+              await this.publish({ ...opts, path: node.url, session })
+            );
+            url.searchParams.set('cache-bypass', uuid.v4());
+            node.url = url.toString();
+          } catch (err) {
+            if (err.code !== 'ENOENT') throw err;
+          }
+        }
+      };
+
+      await Promise.all(nodes.map(visitor));
+    };
+
+    if (publish)
+      report = String(await remark().use(publishLocalFiles).process(report));
+
+    if (watch) {
+      let lock;
+      watcher.add(triggerFile || markdownFile);
+      watcher.on('all', async (event, path) => {
+        if (lock) return;
+        lock = true;
+        try {
+          winston.info(`watcher event: ${event} ${path}`);
+          await this.commentCreate({ ...opts, update: true, watch: false });
+          if (event !== 'unlink' && path === triggerFile)
+            await fs.promises.unlink(triggerFile);
```

Comment:
see above
```suggestion
          if (event !== 'unlink' && path === triggerFile) {
            await fs.promises.unlink(triggerFile);
          }
```

Reply:
Resolving in favor of https://github.com/iterative/cml/pull/1026#discussion_r895175841